libbeat/processors/actions/decode_json_fields.go

```diff
@@ -0,0 +1,93 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/pkg/errors"
+)
+
+type decodeJSONFields struct {
+	Fields []string
+}
+
+func init() {
+	processors.RegisterPlugin("decode_json_fields", configChecked(newDecodeJSONFields,
+		requireFields("fields"), allowedFields("fields", "when")))
+}
+
+func newDecodeJSONFields(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []string `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the decode_json_fields configuration: %s", err)
+	}
+
+	f := decodeJSONFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
+	errs := []string{}
+
+	for _, field := range f.Fields {
+		var output map[string]interface{}
+
+		data, err := event.GetValue(field)
+		if err != nil && errors.Cause(err) != common.ErrKeyNotFound {
+			errs = append(errs, err.Error())
+		}
+
+		text, ok := data.(string)
+		if ok {
+			err := unmarshal([]byte(text), &output)
+			if err != nil {
+				errs = append(errs, err.Error())
+				event.Put(field, text)
+				continue
+			}
+
+			_, addErr := event.Put(field, output)
+			if addErr != nil {
+				errs = append(errs, addErr.Error())
```

Comment:
To be consistent and prevent future bugs this should also `continue` after the error. It doesn't really matter in this case, but if in the future some logic gets added after the location where the error occurred it could cause a bug.